src/contexts/PreferencesContext.tsx

```diff
@@ -189,20 +203,36 @@ export const PreferencesProvider = ({
     );
   }
 
-  function handleDirectoryFavoriteChange(item: DirectoryFavorite) {
-    const existingItemIndex = directoryFavorites.findIndex(
-      dir => dir.name === item.name && dir.fileSharePath === item.fileSharePath
-    );
+  function handleDirectoryFavoriteChange(
+    item: DirectoryFavorite | DirectoryFavorite[]
+  ) {
+    let existingItemIndex;
+    if (Array.isArray(item)) {
```

Comment:
This might be okay, but we want to try to avoid type checks whenever possible, which are generally considered to be [code smells](https://dev.to/mcsee/code-smell-23-instance-type-checking-1i90). They can usually be replaced with polymorphism or some kind of data modeling.